agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestwiseCoverageAgent.java

```diff
@@ -127,8 +127,11 @@ private String handleTestRunStart(Request request, Response response) throws IOE
 			}
 		}
 
-		if (availableTests == null) {
-			availableTests = Collections.emptyList();
+		if (CollectionUtils.isNullOrEmpty(availableTests)) {
+			// the TIA service will reject requests with an empty test list as there is nothing to prioritize.
+			// the more sensible thing to do is to send no test list, in which case the service will simply prioritize
+			// all tests currently known to Teamscale. This is a sensible fallback behaviour
+			availableTests = null;
```

Comment:
Sorry for the confusion. This unfortunately won't work. The PUT .../test-impact always expects a list of tests. Only the GET .../test-impact can handle not having any tests available.

Also like described before, when a list is given by the user of the library I would expect that to be respected aka returning an empty list of impacted tests when the given list was empty as that is a subset of the given tests.

Reply:
In that case we first need to fix this behaviour in TS as that will still reject an empty list. I filed https://jira.cqse.eu/browse/TS-23834 for that